examples/hello_world/src/main.rs

```diff
@@ -0,0 +1,59 @@
+//! A Hello World example application for working with [Gotham](https://gotham.rs).
+
+extern crate futures;
+extern crate gotham;
+extern crate hyper;
+extern crate mime;
+
+use hyper::{Response, StatusCode};
+
+use gotham::http::response::create_response;
+use gotham::start;
+use gotham::state::State;
+
+/// This is an example of a Gotham `Handler` which will always respond regardless of
+/// the `Request` path.
+///
+/// Dealing with breaking up the Request path and dispatching to application code is a one part of
+/// what the full Gotham `Router` offers. This is shown in subsequent examples.
+///
+/// How does a function become a `Handler`?.
+/// We've simply implemented the `Handler` trait for functions that match the signature used here
+/// within Gotham itself.
+pub fn say_hello(state: State) -> (State, Response) {
+    let res = create_response(
+        &state,
+        StatusCode::Ok,
+        Some((String::from("Hello World!").into_bytes(), mime::TEXT_PLAIN)),
+    );
+
+    (state, res)
+}
+
+/// Start a server and call the `Handler` we've defined above for each `Request` we receive.
+pub fn main() {
+    let addr = "127.0.0.1:7878";
+    println!("Listening for requests at http://{}", addr);
+    start(addr, || Ok(say_hello))
```

Comment:
Not a fan of keeping this explicitly as `gotham::start` in code? I think importing the bare function makes it a bit harder to see what's going on.

Reply:
Agree, `gotham::start` more useful, especially for newcomers.

Reply:
fixed